static/js/publisher/release/actions/pendingReleases.js

```diff
@@ -9,17 +9,38 @@ export const PAUSE_PROGRESSIVE_RELEASE = "PAUSE_PROGRESSIVE_RELEASE";
 export const RESUME_PROGRESSIVE_RELEASE = "RESUME_PROGRESSIVE_RELEASE";
 export const CANCEL_PROGRESSIVE_RELEASE = "CANCEL_PROGRESSIVE_RELEASE";
 
-import { getPendingChannelMap } from "../selectors";
-
-export function releaseRevision(
-  revision,
-  channel,
-  progressive,
-  previousRevisions
-) {
-  return {
-    type: RELEASE_REVISION,
-    payload: { revision, channel, progressive, previousRevisions }
+import { getPendingChannelMap, getReleases } from "../selectors";
+
+export function releaseRevision(revision, channel, progressive) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const { revisions } = state;
+
+    const previousRevisions = getReleases(
+      state,
+      revision.architectures,
+      channel
+    )
+      .filter(release => release.revision !== revision.revision)
+      .map(release => revisions[release.revision]);
+
+    if (!progressive) {
+      progressive = {
+        key: null,
```

Comment:
It feels a bit weird to set it to null, so maybe worth adding a comment about when the key is generated?